src/main/java/com/aws/greengrass/secretmanager/SecretManagerIPCAgent.java

```diff
@@ -98,8 +109,11 @@ public GetSecretValueResponse handleRequest(GetSecretValueRequest request) {
 
         @Override
         public CompletableFuture<GetSecretValueResponse> handleRequestAsync(GetSecretValueRequest request) {
-            // TODO: Limit the no of simultaneous requests using a separate threadpool with configurable size
-            return CompletableFuture.supplyAsync(() -> handleRequest(request));
+            if (!Coerce.toBoolean(request.isRefresh())) {
+                return CompletableFuture.completedFuture(handleRequest(request));
```

Comment:
https://github.com/aws-greengrass/aws-greengrass-secret-manager/blob/e23347ce6c5f34498af6fd35724f6daf0089d7cc/src/main/java/com/aws/greengrass/secretmanager/SecretManager.java#L271 looks like it's still possible to do a cloud call when fetching from cache?  any harm in using `cloudCallThreadPool` regardless of `isRefresh`?

Reply:
updated